### PR TITLE
Display a proper message in dropdown list when instructor doesn't have permission to modify courses session. #5180

### LINKED ISSUE #5180: Display a proper message in dropdown list when instructor doesn't have permission to modify courses session.
"No active courses!" is currently shown to the instructor if the instructor have courses under him but doesn't have permissions to modify any of the courses' sessions. A better display message should be shown to instructor with no courses under him and instructors with courses under him but without permission to modify any of the courses' session.

### PR DESCRIPTION
**Pull request by kennho**
Fixes #5180

### CODE REVIEW

File: src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java [hunk 1]
```diff
@@ -290,9 +290,17 @@ public FeedbackSessionsCopyFromModal getCopyFromModal() {
     /**
      * Adds the placeholder option to the list of select options if the list is empty.
      * @param selectOptions list containing all the options
+     * @param courses list of courses that the instructor have
      */
-    private void addPlaceholder(List<ElementTag> selectOptions) {
-        ElementTag placeholder = createOption("No active courses!", "", true);
+    private void addPlaceholder(List<ElementTag> selectOptions, List<CourseAttributes> courses) {
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: It's not very readable why <code>addPlaceholder</code> would need a list of courses. 

I think it will be better if the logic for deciding which message to show is done at a level higher than this method

**Comment by kanghj** [comment]
Remember to add a test too, if none of the existing tests cover this case

**Comment by kennho** [comment]
Okays will add the test for it. I have made some changes. Is this okay? @kanghj

**Comment by kennho** [comment]
Updated with tests @kanghj

File: src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java [hunk 2]
```diff
@@ -160,9 +164,13 @@ private FeedbackSessionsForm buildBasicForm(List<CourseAttributes> courses, Stri
...
-        // adds the placeholder option to courseIdOptions if it is empty
-        if (isSubmitButtonDisabled) {
-            addPlaceholder(courseIdOptions);
+        // Adds a placeholder option to courseIdOptions if it is empty
+        if (courseIdOptions.isEmpty()) {
+            if (courses.isEmpty())
+                addPlaceholder(courseIdOptions, Const.StatusMessages.INSTRUCTOR_NO_ACTIVE_COURSES);
+            else {
+                addPlaceholder(courseIdOptions, Const.StatusMessages.INSTRUCTOR_NO_MODIFY_COURSES_SESSION_PERMISSION);
+            }
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: These two parts are duplicated, so let's extract a method for this

Comment (kennho) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: Hmmm. Actually previously I was doing:

```
private void addPlaceholder(List<ElementTag> selectOptions, List<CourseAttributes> courses) {
   ElementTag placeholder;  
   if (selectOptions.isEmpty()) {
      if (courses.isEmpty()) {
         placeholder = createOption(Const.StatusMessages.INSTRUCTOR_NO_MODIFY_COURSES_SESSION_PERMISSION, "", true);
      } else {
         placeholder = createOption(Const.StatusMessages.INSTRUCTOR_NO_ACTIVE_COURSES, "", true);
      }
   } 
   return placeholder;
}
```

But was rejected in another PR. Can't really think of any other way to do it without it looking too weird @kanghj should I use make that method mentioned above?

Comment (kanghj) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: I don't really like that code you have there too. It's not obvious why <code>addPlaceHolder</code> would need that much information.

File: src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java [hunk 3]
```diff
@@ -160,9 +164,13 @@ private FeedbackSessionsForm buildBasicForm(List<CourseAttributes> courses, Stri
         List<ElementTag> courseIdOptions = getCourseIdOptions(courses, courseIdForNewSession, instructors, newFeedbackSession);
         boolean isSubmitButtonDisabled = courseIdOptions.isEmpty();
         
-        // adds the placeholder option to courseIdOptions if it is empty
-        if (isSubmitButtonDisabled) {
-            addPlaceholder(courseIdOptions);
+        // Adds a placeholder option to courseIdOptions if it is empty
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: Let's remove this comment

File: src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java [hunk 4]
```diff
@@ -115,9 +115,13 @@ private void buildCopyFromModal(List<CourseAttributes> courses, String courseIdF
         
         List<ElementTag> courseIdOptions = getCourseIdOptions(courses, courseIdForNewSession, instructors, newFeedbackSession);
         
-        // adds the default option to courseIdOptions if it is empty
+        // Adds a placeholder option to courseIdOptions if it is empty
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: Here too

File: src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java (hunk 2 above)
Comment (kennho) [inline src/main/java/teammates/ui/controller/InstructorFeedbacksPageData.java]: woah okay

**Comment by kennho** [comment]
Updated with suggested changes @kanghj

File: src/main/java/teammates/common/util/Const.java [hunk 5]
```diff
@@ -1028,6 +1028,8 @@
         public static final String INSTRUCTOR_COURSE_EMPTY = "There are no students in this course. Click <a href=\"%s\">here</a> to enroll students.";
         public static final String INSTRUCTOR_PERSISTENCE_ISSUE = "Account creation is still in progress. Please reload the page"
                 + " after sometime.";
+        public static final String INSTRUCTOR_NO_MODIFY_COURSES_SESSION_PERMISSION = "No permission to modify course's sessions!";
```
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: This sounds like there is only one course. But that's not the case right?

**Comment by damithc** [comment]
Can post screenshots?

**Comment by kennho** [comment]
Sorry for the delay, here are the screenshots @damithc 

Instructor with courses but no permission to modify for all of them
screen shot 2016-04-05 at 12 26 53 pm

Instructor with no courses
screen shot 2016-04-05 at 12 27 42 pm

**Comment by kennho** [comment]
Updated with suggested changes @damithc

File: src/main/java/teammates/common/util/Const.java [hunk 6]
```diff
@@ -1028,7 +1028,7 @@
         public static final String INSTRUCTOR_COURSE_EMPTY = "There are no students in this course. Click <a href=\"%s\">here</a> to enroll students.";
         public static final String INSTRUCTOR_PERSISTENCE_ISSUE = "Account creation is still in progress. Please reload the page"
                 + " after sometime.";
-        public static final String INSTRUCTOR_NO_MODIFY_COURSES_SESSION_PERMISSION = "No permission to modify course's sessions!";
+        public static final String INSTRUCTOR_NO_MODIFY_PERMISSION_FOR_ACTIVE_COURSES_SESSIONS = "No permission to modify course's sessions!";
```
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: I was referring to the variable value rather than name. But my comment applies to both.

File: src/main/java/teammates/common/util/Const.java [hunk 7]
```diff
@@ -1028,6 +1028,8 @@
         public static final String INSTRUCTOR_COURSE_EMPTY = "There are no students in this course. Click <a href=\"%s\">here</a> to enroll students.";
         public static final String INSTRUCTOR_PERSISTENCE_ISSUE = "Account creation is still in progress. Please reload the page"
                 + " after sometime.";
+        public static final String INSTRUCTOR_NO_MODIFY_PERMISSION_FOR_ACTIVE_COURSES_SESSIONS = "No permission to modify course's sessions!";
```
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: `No permission to modify any sessions in un-archived courses`?

File: src/main/java/teammates/common/util/Const.java [hunk 8]
```diff
@@ -1028,6 +1028,8 @@
...
+        public static final String INSTRUCTOR_NO_ACTIVE_COURSES = "No active courses!";
```
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: `No un-archived courses`?

File: src/main/java/teammates/common/util/Const.java (hunk 7 above)
Comment (damithc) [inline src/main/java/teammates/common/util/Const.java]: BTW, this line is too long. Needs wrapping.

**Comment by kennho** [comment]
Updated with suggested changes. @damithc Is it better?
